### PR TITLE
docs: add React examples for board

### PR DESCRIPTION
**Pull request by tomivirkki**
Extracted from https://github.com/vaadin/docs/pull/2632

### CODE REVIEW

File: articles/components/board/index.asciidoc [hunk 1]
```diff
@@ -53,6 +53,27 @@ include::{root}/src/main/java/com/vaadin/demo/component/board/ExampleIndicator.j
...
+[source,tsx]
+----
+include::{root}/frontend/demo/component/board/react/board-basic.tsx[render,tags=snippet,indent=0,group=React]
+----
+endif::[]
+
+ifdef::react[]
+[source,tsx]
+----
+include::{root}/frontend/demo/component/board/react/example-chart.tsx[render,tags=snippet,indent=0,group=React]
```
Comment (web-padawan) [inline articles/components/board/index.asciidoc:67]: This refers to `example-chart.tsx` but the file is named `ExampleChart.tsx`.

File: articles/components/board/index.asciidoc [hunk 2]
```diff
@@ -53,6 +53,27 @@ include::{root}/src/main/java/com/vaadin/demo/component/board/ExampleIndicator.j
...
+----
+endif::[]
+
+ifdef::react[]
+[source,tsx]
+----
+include::{root}/frontend/demo/component/board/react/example-indicator.tsx[render,tags=snippet,indent=0,group=React]
```
Comment (web-padawan) [inline articles/components/board/index.asciidoc:74]: Same as above, the file is named `ExampleIndicator.tsx`.

File: articles/components/board/index.asciidoc (hunk 1 above)
Comment (tomivirkki) [reply articles/components/board/index.asciidoc]: Fixed

File: articles/components/board/index.asciidoc (hunk 2 above)
Comment (tomivirkki) [reply articles/components/board/index.asciidoc]: Fixed

**Review by web-padawan — APPROVED** [review]

**Pull request merged**